### PR TITLE
[Merged by Bors] - feat: port Order.Category.CompleteLatCat

### PR DESCRIPTION
**Pull request by joneugster**
---

- [x] depends on: #5007

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#5007
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Order/Category/CompleteLatCat.lean [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  toFun _ _ f := f.toFun
+  id := @CompleteLatticeHom.id
+  comp := @CompleteLatticeHom.comp
+  hom_ext _ _ _ _ h := FunLike.coe_injective h
+
+instance : LargeCategory.{u} CompleteLatCat :=
+  BundledHom.category CompleteLatticeHom
+
+instance : ConcreteCategory CompleteLatCat :=
+  BundledHom.concreteCategory CompleteLatticeHom
```
Comment (semorrison) [inline Mathlib/Order/Category/CompleteLatCat.lean:62]: please use the `by dsimp [CompleteLatCat]; infer_instance` approach instead.

**Comment by semorrison** [comment]
bors d+

File: Mathlib/Order/Category/CompleteLatCat.lean (hunk 1 above)
Comment (joneugster) [reply Mathlib/Order/Category/CompleteLatCat.lean:62]: ok, did that. But this is mathport output coming from 

```
instance : large_category.{u} CompleteLat := bundled_hom.category complete_lattice_hom
instance : concrete_category CompleteLat := bundled_hom.concrete_category complete_lattice_hom
```

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#5007~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Pull request closed without merging**